### PR TITLE
Update macOS postinstall script to properly build executable paths

### PR DESCRIPTION
**Pull request by bsclifton**
Update postinstall script to properly build executable paths (install path + user data dir name)

Fixes https://github.com/brave/brave-browser/issues/4600

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
1. Download the PKG file (message @bsclifton if unsure where/how to download)
2. Rename it in a format like `Brave Browser Dev-BSC123.pkg` (where `BSC123` is your referral promo code)
3. Run the installer
4. `promoCode` file should be created in `~/Library/BraveSoftware/Brave-Browser-Dev/promoCode`
5. Open file and verify contents are `BSC123` (or whatever your promo code was)

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by bsclifton** [comment]
Gonna let PR builder run so I can get the PKG file and give it a go 😄

**Comment by bsclifton** [comment]
OK finally got this building properly- testing it out now

**Comment by bsclifton** [comment]
Ready for review! I tested this out and it worked great. You can download the .pkg file from build performed with https://staging.ci.brave.com/job/brave-core-build-pr/job/PR-2549/20/

File: build/mac/pkg-scripts/postinstall [hunk 1]
```diff
@@ -6,11 +6,26 @@
...
+    echo "Got promo code: $promoCode"
+  fi
+  appName=$(basename "$1" | sed "s/.pkg//" | sed "s/ /\-/g" | sed "s/-${promoCode}//")
+else
+  echo "Installer path did not match for promo code"
+  promoCode=""
+  appName=$(basename "$1" | sed "s/.pkg//" | sed "s/ /\-/g")
+fi
+
+appNameNoDash=$(echo $appName | sed "s/-/\ /g")
```
Comment (emerick) [inline build/mac/pkg-scripts/postinstall]: It seems like sometimes we're escaping the minus sign and space for sed replacement and sometimes we're not. Wasn't sure if that was intentional or not? I didn't think you needed to escape for space at least, but maybe I'm wrong.

Comment (bsclifton) [reply build/mac/pkg-scripts/postinstall]: oh yeah - good catch. There were various states of this script where the `sed` input had quotes and didn't have quotes. This is an artifact of that. Let me remove the extra escape after testing it and then amend what we have here 😄

**Comment by bsclifton** [comment]
quick note: going to stop the CI for this, as the change I made only affects the `postinstall` script which is not tested as part of CI (also, default builds have signing skipped, which omits the creation of the `.pkg` file)

**Pull request merged**